controllers/storagecluster/provider_server.go

```diff
@@ -1,552 +1,61 @@
-		return reconcile.Result{}, err
-	}
-
-	r.Log.Info("Service create/update succeeded")
-
-	switch instance.Spec.ProviderAPIServerServiceType {
-	case corev1.ServiceTypeLoadBalancer:
-		endpoint := o.getLoadBalancerServiceEndpoint(actualService)
-
-		if endpoint == "" {
-			r.recorder.ReportIfNotPresent(instance, corev1.EventTypeNormal, "Waiting", "Waiting for Ingress on service "+actualService.Name)
-			r.Log.Info("Waiting for Ingress on service", "Service", actualService.Name, "Status", actualService.Status)
-			return reconcile.Result{RequeueAfter: time.Second * 5}, nil
-		}
-
-		instance.Status.StorageProviderEndpoint = fmt.Sprintf("%s:%d", endpoint, ocsProviderServicePort)
-
-	case corev1.ServiceTypeClusterIP:
-		instance.Status.StorageProviderEndpoint = fmt.Sprintf("%s:%d", actualService.Spec.ClusterIP, ocsProviderServicePort)
-
-	default: // Nodeport is the default ServiceType for the provider server
-		nodeAddresses, err := o.getWorkerNodesInternalIPAddresses(r)
-		if err != nil {
-			return reconcile.Result{}, err
-		}
-
-		if len(nodeAddresses) == 0 {
-			err = fmt.Errorf("Could not find any worker nodes")
-			r.Log.Error(err, "Worker nodes count is zero")
-			return reconcile.Result{}, err
-		}
-
-		instance.Status.StorageProviderEndpoint = fmt.Sprintf("%s:%d", nodeAddresses[0], ocsProviderServiceNodePort)
-
-	}
-
-	r.Log.Info("status.storageProviderEndpoint is updated", "Endpoint", instance.Status.StorageProviderEndpoint)
-
+func (o *ocsProviderServer) ensureDeleted(_ *StorageClusterReconciler, _ *ocsv1.StorageCluster) (reconcile.Result, error) {
 	return reconcile.Result{}, nil
 }
-
-func (o *ocsProviderServer) getLoadBalancerServiceEndpoint(service *corev1.Service) string {
-	endpoint := ""
-
-	if len(service.Status.LoadBalancer.Ingress) != 0 {
-		if service.Status.LoadBalancer.Ingress[0].IP != "" {
-			endpoint = service.Status.LoadBalancer.Ingress[0].IP
-		} else if service.Status.LoadBalancer.Ingress[0].Hostname != "" {
-			endpoint = service.Status.LoadBalancer.Ingress[0].Hostname
-		}
-	}
-
-	return endpoint
-}
-
-// getWorkerNodesInternalIPAddresses return slice of Internal IPAddress of worker nodes
-func (o *ocsProviderServer) getWorkerNodesInternalIPAddresses(r *StorageClusterReconciler) ([]string, error) {
-
-	nodes := &corev1.NodeList{}
-
-	err := r.Client.List(context.TODO(), nodes)
-	if err != nil {
-		r.Log.Error(err, "Failed to list nodes")
-		return nil, err
-	}
-
-	nodeAddresses := []string{}
-
-	for i := range nodes.Items {
-		node := &nodes.Items[i]
-		if _, ok := node.ObjectMeta.Labels["node-role.kubernetes.io/worker"]; ok {
-			for _, address := range node.Status.Addresses {
-				if address.Type == corev1.NodeInternalIP {
-					nodeAddresses = append(nodeAddresses, address.Address)
-					break
-				}
-			}
-		}
-	}
-
-	sort.Strings(nodeAddresses)
-
-	return nodeAddresses, nil
-}
-
-func (o *ocsProviderServer) createSecret(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) error {
```

Comment:
I see this wasn't moved to ocsinit, what was this secret used for?

Reply:
I didn't find any use for this secret, and I am not sure why we had it. Nitin mentioned that it was used for the old way of getting tokens